test/python/circuit/test_parameters.py

```diff
@@ -486,12 +495,14 @@ def test_raise_if_assigning_params_not_in_circuit(self):
         y = Parameter("y")
         z = ParameterVector("z", 3)
         qr = QuantumRegister(1)
-        qc = QuantumCircuit(qr)
 
         # test for both `bind_parameters` and `assign_parameters`
         for assign_fun in ["bind_parameters", "assign_parameters"]:
             qc = QuantumCircuit(qr)
             with self.subTest(assign_fun=assign_fun):
+                # TODO: delete once bind_parameters is removed from the codebase
+                warnings.filterwarnings("ignore", category=DeprecationWarning)
+
```

Comment:
This appears to need the manual override because of the `getattr` call changing the stack depth of the deprecated access, right?  We could rewrite the test to avoid needing this, but it's probably cleanest just to leave it the way you have.